pvr.nextpvr/resources/settings.xml

```diff
@@ -29,6 +29,14 @@
             <heading>30002</heading>
           </control>
         </setting>
+        <setting help="" id="hostprotocol" label="" type="string" option="hidden">
+          <level>4</level>
+          <default>http</default>
+          <control format="string" type="edit">
+            <heading></heading>
+            <hidden>true</hidden>
+          </control>
+        </setting>
```

Comment:
Can you explain this? I.e. why do you need a hidden setting?

Reply:
I make it hidden because if it is a toggle the user might think they could simply enable https.  To make this work a user will need to setup  something like an nginix proxy (preferred) or manually configure the NextPVR server with a pfx file.  I figure most users will also use self signed certs for this and Kodi adds the complexity of requiring /system/certs/cacert.pem since I don't want to add a additional option to force override curl security.

Reply:
So if one of those is supplied you set this setting to store the value?

Reply:
No there would be no way to connect.  The user would need to edit the user settings.xml and this would need to be documented.  I did ask @sub3 to consider adding this for the future in the discovery API but I doubt https on the local LAN is important and discovery won't work over a WAN/Internet.

Reply:
Then I would add the documentation as part of this commit.

Reply:
I will be making a markdown file for the entire Matrix update which I have discussed with @Karellen  and this feature will be included but I certainly won't go into how to enable https just how to use https with a note about self-signed cers, but frankly this hidden setting will be one of the lowest priorities since there is no demand, I just didn't like the hard-coded http references in the code..  Do you want the commit removed until later?

Reply:
No, your plan sounds fine. I usually tend to add the documentation with commits. It useful for other contributors then, makes it easier to figure out what is going on.

Reply:
Thanks. I wish I didn't have to consider two communities (NextPVR and Kodi) with their separate forums and wikis.  No matter what or where I document I tend to believe if someone wants to use https they will still post "How do I ..." on one of the forums.

Reply:
Yes, it’s always tricky where to document it.